resolwe_bio/processes/plots/bamliquidator.yml

```diff
@@ -0,0 +1,215 @@
+      label: Matrix file
+      type: basic:file
+      required: False
+      hidden: "analysis_type != 'region'"
+    - name: summary
+      label: Summary file
+      type: list:basic:file:html
+      required: False
+      hidden: "analysis_type != 'bin'"
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+
+      re-save analysis_type {{ analysis_type }}
+
+      {% if regions_gtf and regions_bed %}
+        re-error "Please provide only one type of region file gff or bed file, accordingly to the analysis type."
+      {% endif %}
+
+      {% if regions_gtf and bin_size %}
+        re-error "Please provide only gff file or bin size, accordingly to the analysis type."
+      {% endif %}
+
+      {% if regions_bed and bin_size %}
+        re-error "Please provide only bed file or bin size, accordingly to the analysis type."
+      {% endif %}
+
+      if [{{ analysis_type }} == "bin" ] && [ {{ regions_bed }} ] || [ {{ regions_gtf }} ]; then
+        re-error "In Bin mode, regions bed or gff files can not be used. Please select different analysis type."
+      fi
+
+      if [{{ analysis_type }} == "region" ] && [ {{ regions_bed }} ] || [ {{ bin_size }} ]; then
+        re-error "In Region mode, bed file or bin size can not be used. Please select different analysis type."
+      fi
+
+      if [{{ analysis_type }} == "bed" ] && [ {{ regions_gtf }} ] || [ {{ bin_size }} ]; then
+        re-error "In BED mode, region gtf file or bin size can not be used. Please select different analysis type."
+      fi
+
+      BAM_PARAMS=()
+      BAM_BAI_PARAMS=()
+      ALL_MODE_PAR=()
+
+      BAM_PARAMS+=({% for bam_file in bam %}"{{bam_file.bam.file}}" {% endfor %})
+      BAM_BAI_PARAMS+=({% for bai_file in bam %}"{{bai_file.bai.file}}" {% endfor %})
+      SKIP_PLOT='{% if skip_plot %}--skip_plot {% endif %}'
+      BLACK_LIST='{% if black_list %}--black_list {{black_list|join(' ')}} {% endif %}'
+      SENSE='{% if sense %}--sense {{ sense }} {% endif %}'
+      EXTENSION='{% if extension %}-e {{ extension }} {% endif %}'
+
+      ALL_MODE_PAR+=("{{ cell_type }}" -n {{threads}} --xml_timings ${SKIP_PLOT} ${BLACK_LIST} ${EXTENSION} ${SENSE} -o output)
+
+      echo "Creating folder with bams..."
+      mkdir "{{ cell_type }}"
+      for file in "${BAM_PARAMS[@]}"; do
+        NAME=`basename "$file" .bam`
+        ln -s "$file" "{{ cell_type }}"/"${NAME}".bam
+      done
+
+      for file in "${BAM_BAI_PARAMS[@]}"; do
+        NAME=`basename "$file" .bam.bai`
+        ln -s "$file" "{{ cell_type }}"/"${NAME}".bam.bai
+      done
+
+      if [ {{ analysis_type }} == "bin" ]; then
+        {% if bin_size %}
+          BIN_SIZE={{bin_size}}
+        {% else %}
+          BIN_SIZE=10000
+        {% endif %}
+
+        bamliquidator_batch.py "${ALL_MODE_PAR[@]}" -b ${BIN_SIZE} -f
+        re-checkrc "Bamliquidator failed."
+
+        {% if not skip_plot %}
+          SUMMARY_URL=()
+          SUMMARY_URL+=("{\"file\":\"output/summary.html\"}")
```

Comment:
Is there a reason why is this a list? Probably you can just change field type to `basic:file:html`.